File: face/geomajas-face-dojo/client/src/main/resources/geomajas/action/toolbar/ZoomPreviousAction.js
dojo.provide("geomajas.action.toolbar.ZoomPreviousAction");
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2011 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */
dojo.require("geomajas.action.ToolbarAction");
dojo.require("geomajas.action.toolbar.util.MapViewQueueHandler");

dojo.declare("ZoomPreviousAction", ToolbarAction, {

	/**
	 * @fileoverview Zoom to the previous view on the map.
	 * @class Zoom to the previous view on the map.
	 * @author Pieter De Graef
	 *
	 * @constructor
	 * @extends ToolbarAction
	 * @param id Unique identifier
	 * @param queueHandler Manages the map extent queue.
	 */
	constructor : function (id, queueHandler) {
		/** Unique identifier. */
		this.id = id;

		/** This tool's image representation. */
		this.image = "zoomPreviousIcon";

		/** Reference to the tooltip. */
		this.tooltip = this.tooltipLocale.ZoomPreviousAction;
		
		/** Manager for the map extent queue. */
		this.queueHandler = queueHandler;

		// Initialization:
		dojo.connect(this.queueHandler, "onQueueChange", dojo.hitch(this, "_onMapViewQueueChange"));
		this.setEnabled(false);
	},

	/**
	 * The action's execution function. Calls the MapViewQueueHandler's
	 * zoomToPrevious function.
	 * @param event Standard browser mouseevent.
	 */
	actionPerformed : function (event) {
		this.queueHandler.zoomToPrevious();
	},

	getQueueHandler : function () {
		return this.queueHandler;
	},

	/**
	 * @private 
	 */
	_onMapViewQueueChange : function () {
		if (this.queueHandler.getCurrentQueuePosition() == 0) {
			this.setEnabled(false);
		} else {
			this.setEnabled(true);
		}
	}
});

File: face/geomajas-face-gwt/client/src/main/java/org/geomajas/gwt/client/action/toolbar/ZoomPreviousAction.java
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2013 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */

package org.geomajas.gwt.client.action.toolbar;

import org.geomajas.gwt.client.action.ToolbarAction;
import org.geomajas.gwt.client.i18n.I18nProvider;
import org.geomajas.gwt.client.util.WidgetLayout;
import org.geomajas.gwt.client.widget.MapWidget;

import com.smartgwt.client.widgets.events.ClickEvent;

/**
 * Go back to the previous zoom level.
 * 
 * @author Joachim Van der Auwera
 */
public class ZoomPreviousAction extends ToolbarAction {

	private ZoomQueue zoomQueue;

	public ZoomPreviousAction(MapWidget mapWidget) {
		super(WidgetLayout.iconZoomLast, I18nProvider.getToolbar().zoomPreviousTitle(), I18nProvider
				.getToolbar().zoomPreviousTooltip());
		zoomQueue = ZoomQueue.getZoomQueue(mapWidget);
		zoomQueue.setZoomPreviousAction(this);
	}

	public void onClick(ClickEvent clickEvent) {
		zoomQueue.zoomPrevious();
	}
}
